sources/filesystem/__init__.py

```diff
@@ -11,9 +12,36 @@
     FilesystemConfigurationResource,
     fsspec_from_resource,
 )
+from .readers import ReadersSource, _read_csv, _read_jsonl, _read_parquet
 from .settings import DEFAULT_CHUNK_SIZE
 
 
+@dlt.source(_impl_cls=ReadersSource, spec=FilesystemConfigurationResource)
```

Comment:
I have thought about adding some readers, not just in the example but had no idea, this implementations looks really good!